File: front/src/repository/FilmRepository.ts
const baseUrl = "http://127.0.0.1:8000/v1/films";

import { type Film } from "../model/Film";
import { type CreateFilm } from "../model/CreateFilm";

interface IFilmRepository {
  list_films(): Promise<Film[]>;
  create_film(film: CreateFilm): Promise<Film>;
  update_film(film: Film): Promise<Film>;
  delete_film(id: string): Promise<string>;
}

class FilmRepository implements IFilmRepository {
  private readonly base_url: string;
  constructor(base_url: string) {
    this.base_url = base_url;
  }

  async list_films(): Promise<Film[]> {
    const response = await fetch(this.base_url, {
      method: "GET",
      mode: "cors",
    });
    return await response.json();
  }
  async create_film(film: CreateFilm): Promise<Film> {
    const response = await fetch(this.base_url, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      mode: "cors",
      body: JSON.stringify(film),
    });
    return await response.json();
  }
  async update_film(film: Film): Promise<Film> {
    const response = await fetch(this.base_url, {
      method: "PUT",
      headers: {
        "Content-Type": "application/json",
      },
      mode: "cors",
      body: JSON.stringify(film),
    });
    return await response.json();
  }
  async delete_film(id: string): Promise<string> {
    const response = await fetch(this.base_url + "/" + id, {
      method: "DELETE",
      mode: "cors",
    });
    return await response.json();
  }
}

export const filmRepository = new FilmRepository(baseUrl);

File: desktop/src/main/kotlin/repository/FilmRepository.kt
package repository

import io.ktor.client.*
import io.ktor.client.call.*
import io.ktor.client.request.*
import io.ktor.http.*
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import kotlinx.serialization.json.JsonBuilder
import modal.CreateFilm
import modal.Film


const val BASE_URL = "http://127.0.0.1:8000"

const val FILM_URL = "$BASE_URL/v1/films"

class FilmRepository(private val http: HttpClient) : IFilmRepository {
    override suspend fun getFilms(): List<Film> = withContext(Dispatchers.IO) {
        http.get(FILM_URL).body()
    }

    override suspend fun deleteFilms(film: Film): String =
        withContext(Dispatchers.IO) {
            http.delete("$FILM_URL/${film.id}").body()
        }

    override suspend fun updateFilm(film: Film): Film = withContext(Dispatchers.IO) {
        http.put(FILM_URL) {
            contentType(ContentType.Application.Json)
            setBody(film)
        }.body()
    }

    override suspend fun createFilm(createFilm: CreateFilm): Film = withContext(Dispatchers.IO) {
        http.post(FILM_URL) {
            contentType(ContentType.Application.Json)
            setBody(createFilm)
        }.body()
    }
}